e2e/bitrise.yml

```diff
@@ -443,6 +445,11 @@ workflows:
         - content: |-
             #!/usr/bin/env bash
 
+            if [[ ${XCODE_MAJOR_VERSION} -ge 14 ]]; then
+              echo "Xcode 14+ does not output dSYM files "
```

Comment:
I might be behind the times here, but is this the most accurate message that we could convey?  I can't find anything online that says that Apple has deprecated dSYM and I believe it's still the preferred format for capturing debugging symbols.  I haven't investigated closely, so maybe not.  I'm curious if there are more nuanced conditions under which dSYM files wouldn't be generated.  Does Xcode 14 specify different build flags than prior versions?

Reply:
Good question, my assumption is that this is a side effect of the bitcode deprecation
From the [release notes](https://developer.apple.com/documentation/Xcode-Release-Notes/xcode-14-release-notes)
> Deprecations
Starting with Xcode 14, bitcode is no longer required for watchOS and tvOS applications, and the App Store no longer accepts bitcode submissions from Xcode 14.

> Xcode no longer builds bitcode by default and generates a warning message if a project explicitly enables bitcode: “Building with bitcode is deprecated. Please update your project and/or target settings to disable bitcode.” The capability to build with bitcode will be removed in a future Xcode release. IPAs that contain bitcode will have the bitcode stripped before being submitted to the App Store. Debug symbols for past bitcode submissions remain available for download. (86118779)

Reply:
My assumption is that dSYM files are a side effect of bitcode.

Enabling bitcode allows Apple to post process your build and create slices of your product for each processor variant. So for each type of processor, they create a build for that explicit device type. I believe the impetus for this was to reduce the overall binary size. dSYM files contain the mappings between your original build and the post processed build so that crash reports can be mapped back to their original symbols for debugging purposes. 

When you compile with bitcode enabled, you need the dSYM files to desymbolicate your crash reports. 
In cases where bitcode is disabled, you do not need the dSYM files because you always know which symbols the OS used with your build.